flagsmith/analytics.py

```diff
@@ -0,0 +1,67 @@
+import copy
+import logging
+from collections import UserDict
+from datetime import datetime
+
+import requests
+from requests_futures.sessions import FuturesSession
+
+logger = logging.getLogger(__name__)
+
+ANALYTICS_ENDPOINT = "analytics/flags/"
+
+# Used to control how often we send data(in seconds)
+ANALYTICS_TIMER = 10
+
+session = FuturesSession(max_workers=4)
+
+
+class AnalyticsProcessor:
+    """
+    AnalyticsProcessor is used to track how often individual Flags are evaluated within the Flagsmith SDK
+    docs: https://docs.flagsmith.com/advanced-use/flag-analytics
+    """
+
+    def __init__(self, environment_key: str, base_api_url: str, timeout: int = 3):
+        """
+        Initialise Flagsmith environment.
+
+        :param environment_key: environment key obtained from the Flagsmith UI
+        :param  base_api_url:  base api url to override when using self hosted version
+        :param timeout: used to tell requests to stop waiting for a response after a given number of seconds
+        """
+        self.analytics_endpoint = base_api_url + ANALYTICS_ENDPOINT
+        self.environment_key = environment_key
+        self._last_flushed = datetime.now()
+        self.analytics_data = {}
+        self.timeout = timeout
+        super().__init__()
+
+    def flush(self):
+        """
+        Sends all the collected data to the api asynchronously and resets the timer
```

Comment:
I don't know that this _will_ be sent asyncronously will it? Since the rest of our API request is synchronous.